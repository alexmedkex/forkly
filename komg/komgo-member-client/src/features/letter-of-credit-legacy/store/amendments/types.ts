import { ImmutableMap, stringOrNull } from '../../../../utils/types'
import { ILCAmendment } from '@komgo/types'
import { List, Map } from 'immutable'

export interface LetterOfCreditAmendmentStateProperties {
  byStaticId: Map<string, ILCAmendment>
  staticIds: List<string>
  error: stringOrNull
}

export type LetterOfCreditAmendmentState = ImmutableMap<LetterOfCreditAmendmentStateProperties>

export enum LetterOfCreditAmendmentActionType {
  SUBMIT_AMENDMENT_REQUEST = '@@letter-of-credit-amendments/SUBMIT_AMENDMENT_REQUEST',
  SUBMIT_AMENDMENT_SUCCESS = '@@letter-of-credit-amendments/SUBMIT_AMENDMENT_SUCCESS',
  SUBMIT_AMENDMENT_FAILURE = '@@letter-of-credit-amendments/SUBMIT_AMENDMENT_FAILURE',
  GET_AMENDMENT_REQUEST = '@@letter-of-credit-amendments/GET_AMENDMENT_REQUEST',
  GET_AMENDMENT_SUCCESS = '@@letter-of-credit-amendments/GET_AMENDMENT_SUCCESS',
  GET_AMENDMENT_FAILURE = '@@letter-of-credit-amendments/GET_AMENDMENT_FAILURE',
  ISSUE_AMENDMENT_REQUEST = '@@letter-of-credit-amendments/ISSUE_AMENDMENT_REQUEST',
  ISSUE_AMENDMENT_SUCCESS = '@@letter-of-credit-amendments/ISSUE_AMENDMENT_SUCCESS',
  ISSUE_AMENDMENT_FAILURE = '@@letter-of-credit-amendments/ISSUE_AMENDMENT_FAILURE',
  REJECT_AMENDMENT_REQUEST = '@@letter-of-credit-amendments/REJECT_AMENDMENT_REQUEST',
  REJECT_AMENDMENT_SUCCESS = '@@letter-of-credit-amendments/REJECT_AMENDMENT_SUCCESS',
  REJECT_AMENDMENT_FAILURE = '@@letter-of-credit-amendments/REJECT_AMENDMENT_FAILURE'
}
