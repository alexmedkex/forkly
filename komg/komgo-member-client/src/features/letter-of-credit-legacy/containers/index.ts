import LetterOfCreditApplication from './LetterOfCreditApplication'
import LetterOfCreditDashboard from './LetterOfCreditDashboard'
import LetterOfCreditView from './LetterOfCreditView'
import LetterOfCreditPresentation from './presentation/LetterOfCreditPresentation'
import LetterOfCreditPresentationHistory from './presentation/LetterOfCreditPresentationHistory'
import LetterOfCreditPresentationDetails from './presentation/LetterOfCreditPresentationDetails'
import LetterOfCreditPresentationFeedback from './presentation/LetterOfCreditPresentationFeedback'
import LetterOfCreditAuthorization from './LetterOfCreditAuthorization'

export {
  LetterOfCreditApplication,
  LetterOfCreditDashboard,
  LetterOfCreditView,
  LetterOfCreditPresentation,
  LetterOfCreditPresentationHistory,
  LetterOfCreditPresentationDetails,
  LetterOfCreditPresentationFeedback,
  LetterOfCreditAuthorization
}
