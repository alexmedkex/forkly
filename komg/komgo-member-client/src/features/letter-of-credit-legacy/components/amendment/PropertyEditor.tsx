import * as React from 'react'
import { Segment, Dropdown, Icon } from 'semantic-ui-react'
import { ILetterOfCredit } from '../../types/ILetterOfCredit'
import { FormikContext, connect } from 'formik'
import { fieldToLabel } from '../../constants/fieldsByStep'
import { LetterOfCreditAmendmentContext, findDiffsByTypes } from '../../containers/CreateAmendment'
import { selectInputForField } from '../../utils/selectInputForField'
import { violetBlue } from '../../../../styles/colors'
import { FIELD_ERROR_CLASSNAME } from '../../constants'
import { ILCAmendmentBase } from '@komgo/types'

interface PropertyEditorOwnProps {
  index: number
  field: keyof ILetterOfCredit | ''
  options: any[]
}

export type PropertyEditorProps = PropertyEditorOwnProps & {
  formik: FormikContext<ILCAmendmentBase>
}

export const pathToKey = (s: string): string => s.split('/')[1]
export const keyToPath = (s: string) => `/${s}`
export const selectDefaultValue = (s: string) => {
  switch (s) {
    case 'number':
      return 0
    case 'boolean':
      return false
    default:
      return ''
  }
}

// ATM is related to the LC
export const PropertyEditor: React.FC<PropertyEditorProps> = ({
  field,
  options,
  index,
  formik: { values, setFieldValue, errors }
}) => {
  const fieldDiff = values.diffs.find(v => v.path === keyToPath(field))

  const fieldHasError = Object.keys(errors).includes(field)

  const onFieldChange = (value: string) => {
    setFieldValue('diffs', values.diffs.map(v => (pathToKey(v.path) === field ? { ...v, value } : v)))
  }

  const lcDiffs = findDiffsByTypes(values.diffs, ['ILC'])
  const tradeDiffs = findDiffsByTypes(values.diffs, ['ITrade', 'ICargo'])

  return (
    <Segment>
      <div style={{ display: 'flex', justifyContent: 'space-between', paddingBottom: '10px' }}>
        <div>Field</div>
        <Icon
          name="close"
          link={true}
          style={{ color: `${violetBlue}` }}
          onClick={() => setFieldValue('diffs', [...tradeDiffs, ...lcDiffs.filter((_, idx) => index !== idx)])}
        />
      </div>
      <div style={{ maxWidth: '350px' }}>
        <LetterOfCreditAmendmentContext.Consumer>
          {(letterOfCredit: ILetterOfCredit) => (
            <Dropdown
              fluid={true}
              options={options.map(o => ({ value: o, content: fieldToLabel(o), text: fieldToLabel(o) }))}
              selection={true}
              value={field}
              className={fieldHasError ? FIELD_ERROR_CLASSNAME : ''}
              placeholder="Select an amendable field..."
              onChange={(_, { value: pathValue }: { value: string }) =>
                setFieldValue('diffs', [
                  ...tradeDiffs,
                  ...lcDiffs.map(
                    (l, idx) =>
                      index === idx
                        ? {
                            op: 'replace',
                            path: keyToPath(pathValue),
                            oldValue: letterOfCredit[pathValue],
                            value: selectDefaultValue(typeof letterOfCredit[pathValue]),
                            type: 'ILC'
                          }
                        : l
                  )
                ])
              }
            />
          )}
        </LetterOfCreditAmendmentContext.Consumer>
        {!!field && (
          <div style={{ display: 'flex', paddingTop: '15px' }}>
            <div style={{ width: '100%', paddingLeft: '10px' }}>
              Current
              {selectInputForField(field, fieldDiff)}
            </div>
            <div style={{ width: '100%', paddingLeft: '10px' }}>
              New
              {selectInputForField(field, fieldDiff, onFieldChange)}
            </div>
          </div>
        )}
      </div>
    </Segment>
  )
}

export default connect<PropertyEditorOwnProps, ILCAmendmentBase>(PropertyEditor)
