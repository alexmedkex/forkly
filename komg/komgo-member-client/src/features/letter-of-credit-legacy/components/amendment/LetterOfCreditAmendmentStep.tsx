import * as React from 'react'
import { Segment } from 'semantic-ui-react'
import { Field, connect, FormikContext } from 'formik'

import { CapitalizedHeader } from '../CapitalizedHeader'
import PropertyEditor, { pathToKey } from './PropertyEditor'
import { ILetterOfCredit } from '../../types/ILetterOfCredit'
import SimpleButton from '../../../../components/buttons/SimpleButton'
import { violetBlue, buttonGrey } from '../../../../styles/colors'
import { IDiff, ILCAmendmentBase } from '@komgo/types'
import { findDiffsByTypes } from '../../containers/CreateAmendment'

interface LetterOfCreditAmendmentOwnProps {
  disabled?: boolean
}

interface LetterOfCreditAmendmentState {
  newUpdate: boolean
}

export type LetterOfCreditAmendmentProps = LetterOfCreditAmendmentOwnProps & {
  formik: FormikContext<ILCAmendmentBase>
}

export const amendableFields: Array<Partial<keyof ILetterOfCredit>> = [
  'feesPayableBy',
  'invoiceRequirement',
  'billOfLadingEndorsement',
  'currency',
  'amount',
  'expiryDate',
  'availableWith',
  'partialShipmentAllowed',
  'transhipmentAllowed',
  'LOI'
]

export class LetterOfCreditAmendmentStep extends React.Component<
  LetterOfCreditAmendmentProps,
  LetterOfCreditAmendmentState
> {
  constructor(props: LetterOfCreditAmendmentProps) {
    super(props)
  }
  render() {
    const {
      formik: { values, setFieldValue }
    } = this.props

    const letterOfCreditAmendments = findDiffsByTypes(values.diffs, ['ILC'])

    const moreUpdatesPossible = amendableFields.length !== letterOfCreditAmendments.length

    return (
      <>
        <CapitalizedHeader content="Updates" block={true} />
        <Segment basic={true}>
          <Field name="diffs" render={() => generatePropertyEditors(letterOfCreditAmendments)} />

          <SimpleButton
            type="button"
            disabled={!moreUpdatesPossible}
            color={moreUpdatesPossible ? violetBlue : buttonGrey}
            onClick={() =>
              setFieldValue('diffs', [
                ...values.diffs,
                { op: 'replace', value: '', oldValue: '', path: '', type: 'ILC' }
              ])
            }
          >
            + Add field
          </SimpleButton>
        </Segment>
      </>
    )
  }
}

const generatePropertyEditors = (lcDiffs: IDiff[]) =>
  lcDiffs
    .filter(diff => amendableFields.includes(pathToKey(diff.path) as keyof ILetterOfCredit) || diff.path === '')
    .map((diff, idx) => (
      <PropertyEditor
        key={idx}
        index={idx}
        options={amendableFields.filter(field => selectValidDropdownOptions(lcDiffs, diff, field))}
        field={(pathToKey(diff.path) as keyof ILetterOfCredit) || ''}
      />
    ))

const selectValidDropdownOptions = (allDiffs: IDiff[], currentDiff: IDiff, field: string) =>
  !allDiffs.map(diff => pathToKey(diff.path)).includes(field) || pathToKey(currentDiff.path) === field

export default connect<LetterOfCreditAmendmentOwnProps, ILCAmendmentBase>(LetterOfCreditAmendmentStep)
