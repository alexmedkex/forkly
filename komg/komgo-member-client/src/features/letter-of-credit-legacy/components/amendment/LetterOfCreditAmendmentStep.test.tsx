import * as React from 'react'
import * as renderer from 'react-test-renderer'
import LetterOfCreditAmendmentStep, { amendableFields } from './LetterOfCreditAmendmentStep'
import { mount, ReactWrapper } from 'enzyme'
import { FormikProvider } from 'formik'
import { fakeFormikContext } from '../../../../store/common/faker'
import { LetterOfCreditAmendmentContext } from '../../containers/CreateAmendment'
import { fakeLetterOfCredit, fakeLetterOfCreditDiff } from '../../utils/faker'
import { PropertyEditor, keyToPath } from './PropertyEditor'
import { Icon } from 'semantic-ui-react'
import { INVOICE_REQUIREMENT_OPTIONS } from '../../constants'
import { ILCAmendmentBase, Currency } from '@komgo/types'

const lc = fakeLetterOfCredit()

const lcAmendments = [fakeLetterOfCreditDiff()]

const setFieldValue = jest.fn()

const formikContext = fakeFormikContext<ILCAmendmentBase>(
  {
    diffs: lcAmendments,
    lcStaticId: lc._id,
    lcReference: lc.reference,
    version: 1
  },
  { setFieldValue }
)

describe('LetterOfCreditAmendmentStep', () => {
  let wrapper: ReactWrapper
  beforeEach(() => {
    jest.resetAllMocks()
    wrapper = mount(
      <LetterOfCreditAmendmentContext.Provider value={lc}>
        <FormikProvider value={formikContext}>
          <LetterOfCreditAmendmentStep />
        </FormikProvider>
      </LetterOfCreditAmendmentContext.Provider>
    )
  })
  it('shows a create field link', () => {
    expect(wrapper.find('button').text()).toEqual('+ Add field')
  })
  it('calls setFieldValue with the right arguments when you click create amendment button', () => {
    expect(wrapper.find(PropertyEditor).length).toEqual(1)

    wrapper.find('button').simulate('click')

    expect(setFieldValue).toHaveBeenCalledWith('diffs', [
      { oldValue: 'OTHER', op: 'replace', path: keyToPath('feesPayableBy'), value: 'SPLIT', type: 'ILC' },
      { oldValue: '', op: 'replace', path: '', value: '', type: 'ILC' }
    ])
  })
  it('still shows the add amendment button when you click create amendment button', () => {
    wrapper.find('button').simulate('click')

    expect(wrapper.find('button').length).toEqual(1)
  })
  it('hides the new update box when you click close on the new PropertyEditor', () => {
    wrapper.find('button').simulate('click')

    wrapper
      .find(PropertyEditor)
      .last()
      .find(Icon)
      .find({ name: 'close' })
      .simulate('click')

    expect(wrapper.find(PropertyEditor).length).toEqual(1)
  })
  it('Gives the update box all of the options', () => {
    const props = wrapper.find(PropertyEditor).prop('options')

    expect(props).toEqual(amendableFields)
  })
  it('omits options already in other PropertyEditors', () => {
    const customFormikContext = {
      ...formikContext,
      values: {
        lcStaticId: lc._id,
        lcReference: lc.reference,
        version: 1,
        diffs: [...lcAmendments, fakeLetterOfCreditDiff({ path: keyToPath('availableWith') })]
      }
    }
    wrapper = mount(
      <LetterOfCreditAmendmentContext.Provider value={lc}>
        <FormikProvider value={customFormikContext}>
          <LetterOfCreditAmendmentStep />
        </FormikProvider>
      </LetterOfCreditAmendmentContext.Provider>
    )

    const PropertyEditors = wrapper.find(PropertyEditor)

    expect(PropertyEditors.length).toEqual(2)

    expect(PropertyEditors.first().prop('options')).toEqual(
      amendableFields.filter(f => keyToPath(f) !== customFormikContext.values.diffs[1].path)
    )

    expect(PropertyEditors.last().prop('options')).toEqual(
      amendableFields.filter(f => keyToPath(f) !== customFormikContext.values.diffs[0].path)
    )
  })
  it('disables the create amendment button if all amendments in list are completed', () => {
    const allAmendments = amendableFields.map(f => fakeLetterOfCreditDiff({ path: keyToPath(f) }))

    wrapper = mount(
      <LetterOfCreditAmendmentContext.Provider value={lc}>
        <FormikProvider
          value={fakeFormikContext<ILCAmendmentBase>({
            diffs: allAmendments,
            lcStaticId: lc._id,
            lcReference: lc.reference,
            version: 1
          })}
        >
          <LetterOfCreditAmendmentStep />
        </FormikProvider>
      </LetterOfCreditAmendmentContext.Provider>
    )

    expect(wrapper.find('button').prop('disabled')).toEqual(true)
  })
  it('gives each PropertyEditor an index', () => {
    const allAmendments = amendableFields.map(f => fakeLetterOfCreditDiff({ path: keyToPath(f) }))

    wrapper = mount(
      <LetterOfCreditAmendmentContext.Provider value={lc}>
        <FormikProvider
          value={fakeFormikContext<ILCAmendmentBase>({
            diffs: allAmendments,
            lcStaticId: lc._id,
            lcReference: lc.reference,
            version: 1
          })}
        >
          <LetterOfCreditAmendmentStep />
        </FormikProvider>
      </LetterOfCreditAmendmentContext.Provider>
    )

    expect(wrapper.find(PropertyEditor).length).toEqual(allAmendments.length)

    expect(
      wrapper
        .find(PropertyEditor)
        .at(3)
        .prop('index')
    ).toEqual(3)
  })
  it('hides any differences which are not in the list of accepted differences', () => {
    wrapper = mount(
      <LetterOfCreditAmendmentContext.Provider value={lc}>
        <FormikProvider
          value={fakeFormikContext<ILCAmendmentBase>({
            lcStaticId: lc._id,
            lcReference: lc.reference,
            version: 1,
            diffs: [fakeLetterOfCreditDiff({ value: 'hi', oldValue: 'hey', path: keyToPath('expiryPlace') })]
          })}
        >
          <LetterOfCreditAmendmentStep />
        </FormikProvider>
      </LetterOfCreditAmendmentContext.Provider>
    )

    expect(wrapper.find(PropertyEditor).length).toEqual(0)
  })
  it('renders a LetterOfCreditAmendmentStep with two amendments', () => {
    expect(
      renderer
        .create(
          <LetterOfCreditAmendmentContext.Provider value={lc}>
            <FormikProvider
              value={fakeFormikContext<ILCAmendmentBase>({
                lcStaticId: lc._id,
                lcReference: lc.reference,
                version: 1,
                diffs: [
                  fakeLetterOfCreditDiff({
                    value: INVOICE_REQUIREMENT_OPTIONS.EXHAUSTIVE,
                    oldValue: INVOICE_REQUIREMENT_OPTIONS.SIMPLE,
                    path: keyToPath('invoiceRequirement')
                  }),
                  fakeLetterOfCreditDiff({
                    value: Currency.EUR,
                    oldValue: Currency.USD,
                    path: keyToPath('currency')
                  })
                ]
              })}
            >
              <LetterOfCreditAmendmentStep />
            </FormikProvider>
          </LetterOfCreditAmendmentContext.Provider>
        )
        .toJSON()
    ).toMatchSnapshot()
  })
})
