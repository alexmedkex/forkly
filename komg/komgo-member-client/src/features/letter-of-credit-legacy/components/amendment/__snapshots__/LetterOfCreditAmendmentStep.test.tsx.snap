// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterOfCreditAmendmentStep renders a LetterOfCreditAmendmentStep with two amendments 1`] = `
Array [
  .c0.c0.c0 {
  text-transform: uppercase;
  border: none;
  background-color: #f2f5f8;
}

<div
    className="ui block header c0"
  >
    Updates
  </div>,
  .c0 {
  color: #5700b5;
  border: none;
  cursor: pointer;
  background-color: #ffffff;
}

.c0:focus {
  outline-color: #ffffff;
}

<div
    className="ui basic segment"
  >
    <div
      className="ui segment"
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
            "paddingBottom": "10px",
          }
        }
      >
        <div>
          Field
        </div>
        <i
          aria-hidden="true"
          className="close link icon"
          onClick={[Function]}
          style={
            Object {
              "color": "#5700b5",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "maxWidth": "350px",
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui fluid selection dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Invoice requirement
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
          >
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Fees payable by
              </span>
            </div>
            <div
              aria-checked={true}
              aria-selected={true}
              className="active selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Invoice requirement
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Bill of lading endorsement
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Opening amount
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Date of expiry
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Available with
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Partial shipment allowed
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Transhipment allowed
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                LOI template
              </span>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "paddingTop": "15px",
            }
          }
        >
          <div
            style={
              Object {
                "paddingLeft": "10px",
                "width": "100%",
              }
            }
          >
            Current
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <div
                className="ui disabled radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                style={
                  Object {
                    "flexGrow": 1,
                    "padding": "5px",
                  }
                }
              >
                <input
                  checked={false}
                  className="hidden"
                  disabled={true}
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={-1}
                  type="radio"
                />
                <label>
                  Exhaustive
                </label>
              </div>
              <div
                className="ui checked disabled radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                style={
                  Object {
                    "flexGrow": 1,
                    "padding": "5px",
                  }
                }
              >
                <input
                  checked={true}
                  className="hidden"
                  disabled={true}
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={-1}
                  type="radio"
                />
                <label>
                  Simple
                </label>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "paddingLeft": "10px",
                "width": "100%",
              }
            }
          >
            New
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <div
                className="ui checked radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                style={
                  Object {
                    "flexGrow": 1,
                    "padding": "5px",
                  }
                }
              >
                <input
                  checked={true}
                  className="hidden"
                  disabled={false}
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={0}
                  type="radio"
                />
                <label>
                  Exhaustive
                </label>
              </div>
              <div
                className="ui radio checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                style={
                  Object {
                    "flexGrow": 1,
                    "padding": "5px",
                  }
                }
              >
                <input
                  checked={false}
                  className="hidden"
                  disabled={false}
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex={0}
                  type="radio"
                />
                <label>
                  Simple
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ui segment"
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
            "paddingBottom": "10px",
          }
        }
      >
        <div>
          Field
        </div>
        <i
          aria-hidden="true"
          className="close link icon"
          onClick={[Function]}
          style={
            Object {
              "color": "#5700b5",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "maxWidth": "350px",
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui fluid selection dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Currency
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
          >
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Fees payable by
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Bill of lading endorsement
              </span>
            </div>
            <div
              aria-checked={true}
              aria-selected={true}
              className="active selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Currency
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Opening amount
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Date of expiry
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Available with
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Partial shipment allowed
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Transhipment allowed
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                LOI template
              </span>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "paddingTop": "15px",
            }
          }
        >
          <div
            style={
              Object {
                "paddingLeft": "10px",
                "width": "100%",
              }
            }
          >
            Current
            <div
              aria-disabled={true}
              aria-expanded={false}
              className="ui disabled compact selection dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="listbox"
              style={
                Object {
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <div
                aria-live="polite"
                className="text"
                role="alert"
              >
                USD
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    AED
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    CHF
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    EUR
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    GBP
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    JPY
                  </span>
                </div>
                <div
                  aria-checked={true}
                  aria-selected={true}
                  className="active selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    USD
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "paddingLeft": "10px",
                "width": "100%",
              }
            }
          >
            New
            <div
              aria-disabled={false}
              aria-expanded={false}
              className="ui compact selection dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="listbox"
              style={
                Object {
                  "width": "100%",
                }
              }
              tabIndex={0}
            >
              <div
                aria-live="polite"
                className="text"
                role="alert"
              >
                EUR
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    AED
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    CHF
                  </span>
                </div>
                <div
                  aria-checked={true}
                  aria-selected={true}
                  className="active selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    EUR
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    GBP
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    JPY
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    USD
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c0"
      color="#5700b5"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      + Add field
    </button>
  </div>,
]
`;
